src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/WasmTestMessagesProcessor.cs

```diff
@@ -224,6 +225,12 @@ private void ProcessMessage(string message, bool isError = false)
         if (line.StartsWith("WASM EXIT"))
         {
             _logger.LogDebug("Reached wasm exit");
+
+            // until WASI can work with unix exit code https://github.com/WebAssembly/wasi-cli/pull/44
+            if (line.Length > 10)
+            {
+                ForwardedExitCode = int.Parse(line.Substring(10));
```

Comment:
nit: add a comment that explains we have the exit code in the string and we strip the prefix